Mailer/TwigMailer.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the FOSUserBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\UserBundle\Mailer;
+
+use FOS\UserBundle\Model\UserInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * @author Christophe Coevoet <stof@notk.org>
+ */
+class TwigMailer implements MailerInterface
+{
+    /**
+     * @var \Swift_Mailer
+     */
+    protected $mailer;
+
+    /**
+     * @var UrlGeneratorInterface
+     */
+    protected $router;
+
+    /**
+     * @var \Twig_Environment
+     */
+    protected $twig;
+
+    /**
+     * @var array
+     */
+    protected $parameters;
+
+    /**
+     * TwigMailer constructor.
+     *
+     * @param \Swift_Mailer         $mailer
+     * @param UrlGeneratorInterface $router
+     * @param \Twig_Environment     $twig
+     * @param array                 $parameters
+     */
+    public function __construct($mailer, UrlGeneratorInterface $router, \Twig_Environment $twig, array $parameters)
```

Comment:
$mailer Should be typed as \Swift_Mailer

Reply:
No it should not =) With a type hint it would not be possible to pass `null` value. Old `TwigSwiftMailer` had that type hint. I could not remove it from there because it would be BC break. So - i created a new class `TwigMailer`, a parent for old `TwigSwiftMailer`

Reply:
Ok, what's about
```php
if(null !== $mailer && !($mailer instanceof \Swift_Mailer)
{
    // throw ....
}
```
?

Currently you can pass anything as `$mailer`... :D

Reply:
It turned out that there are more then one mailer! I found this out in @stof's comment last year in https://github.com/FriendsOfSymfony/FOSUserBundle/issues/2562#issuecomment-333584292 =) That is a reason why `symfony/swiftmailer-bundle` is not required dependency, btw. So, it is not necessary, guess

Reply:
I think that there is no another mailer which uses the Swift_Message class (which comes from the swiftmailer package)
https://github.com/covex-nn/FOSUserBundle/blob/b3000d0088536a1384b7057aa0888a9caa42a45e/Mailer/TwigMailer.php#L115

Reply:
@Devtronic actualy, i agree with you, and i am on your side! May be `symfony/swiftmailer-bundle` should be in `require` section of composer.json! Create a PR and i will be very glad if it will be accepted, then i will close this one =)

But, `Swift_Message` comes from `swiftmailer/swiftmailer` and it is not a bundle. So, it is possible to create a bundle, that provides `mailer` service and uses `Swift_Message` for `send` method at the same time.